mpi-proxy-split/mpi-wrappers/mpi_request_wrappers.cpp

```diff
@@ -393,6 +396,38 @@ USER_DEFINED_WRAPPER(int, Request_get_status, (MPI_Request) request,
   return retval;
 }
 
+USER_DEFINED_WRAPPER(int, Cancel, (MPI_Request *) request)
+{
+  int retval;
+  DMTCP_PLUGIN_DISABLE_CKPT();
+  MPI_Request realRequest = VIRTUAL_TO_REAL_REQUEST(*request);
+  JUMP_TO_LOWER_HALF(lh_info.fsaddr);
+  if( realRequest == MPI_REQUEST_NULL ){
```

Comment:
Isn't the `if` branch superfluous?  Won't the lower half simply return with success if the request is MPI_REQUEST_NULL?
If so, why should MANA duplicate the logic of MPI?